stream_alert/rule_processor/alert_forward.py

```diff
@@ -0,0 +1,199 @@
+"""
+Copyright 2017-present, Airbnb Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from datetime import datetime
+import json
+import os
+import time
+
+import backoff
+import boto3
+from botocore.exceptions import ClientError
+
+from stream_alert.shared import backoff_handlers
+from stream_alert.rule_processor import LOGGER
+
+
+class AlertForwarder(object):
+    """Sends alerts to the Alert Processor and the alerts Dynamo table."""
+    # TODO: Do not send to Alert Processor after Alert Merger is implemented
+    BACKOFF_MAX_RETRIES = 6
+
+    def __init__(self, env):
+        """Initialize the Forwarder with the boto3 clients and resource names.
+
+        Args:
+            env (dict): loaded dictionary containing environment information
+        """
+        self.env = env
+        self.client_dynamo = boto3.client('dynamodb', region_name=self.env['lambda_region'])
+        self.client_lambda = boto3.client('lambda', region_name=self.env['lambda_region'])
+        self.function = os.environ['ALERT_PROCESSOR']
+        self.table = os.environ['ALERT_TABLE']
+
+        # Keep track of unprocessed items when retrying batch_write_item()
+        self.unprocessed_items = None
+
+    def _send_to_lambda(self, alerts):
+        """DEPRECATED: Invoke Alert Processor directly
+
+        Sends a message to the alert processor with the following JSON format:
+            {
+                "record": record,
```

Comment:
this comment is outdated, the schema is:

```
                'record': record,
                'rule_name': rule.rule_name,
                'rule_description': rule.rule_function.__doc__ or DEFAULT_RULE_DESCRIPTION,
                'log_source': str(payload.log_source),
                'log_type': payload.type,
                'outputs': rule.outputs,
                'source_service': payload.service(),
                'source_entity': payload.entity,
                'context': rule.context}
```

Reply:
I'll go ahead and change it, but this entire method will be deleted in a few days :) (rule processor will no longer invoke alert processor at all)